### PR TITLE
feat(suite): dispatch coinjoin modals naturally

### PR DESCRIPTION
**Pull request by dahaca**
## Description

dispatch the coinjoin modals naturally

## Related Issue

follow up for https://github.com/trezor/trezor-suite/issues/6117

### CODE REVIEW

File: packages/suite/src/actions/wallet/coinjoinClientActions.ts [hunk 1]
```diff
@@ -101,12 +102,31 @@ const onCoinjoinRoundChanged =
...
+                        relatedAccountKey,
+                    }),
+                );
             }
 
             if (round.phase === RoundPhase.Ended) {
-                // TODO: close critical phase modal
+                dispatch(closeModal());
+
+                const { session } = coinjoinAccountsWithSession[0];
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/coinjoinClientActions.ts]:
```
const isSessionCompleted = coinjoinAccountsWithSession.find(a => a.session?.signedRounds.length === a.session?.maxRounds);
```

Comment (dahaca) [reply packages/suite/src/actions/wallet/coinjoinClientActions.ts]: ba9cf522a8988434e8742a84b06a927a3c4e867e

File: packages/suite/src/actions/wallet/coinjoinClientActions.ts [hunk 2]
```diff
@@ -101,12 +102,32 @@ const onCoinjoinRoundChanged =
...
         // critical actions should be triggered only once
         if (phaseChanged) {
+            const relatedAccountKey = coinjoinAccountsWithSession[0].key; // since all accounts share the round, any key can be used
+
             if (round.phase === RoundPhase.ConnectionConfirmation) {
-                // TODO: open critical phase modal
+                dispatch(
+                    openModal({
+                        type: 'critical-coinjoin-phase',
+                        relatedAccountKey,
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/coinjoinClientActions.ts:158]: you dont need to send `relatedAccountKey` to modal, to iterate over coinjoin accounts again only to get the phase.

just send phase from here

Comment (dahaca) [reply packages/suite/src/actions/wallet/coinjoinClientActions.ts:158]: The modal is tracking round changes, so it would have to be dispatched with new phase every time, instead of dispatching once and then it accesses the phase itself :)

**Review by komret — APPROVED** [review]

**Comment by komret** [comment]
Screenshot 2022-10-26 at 14 01 04
The check icons for rounds that haven't run yet are confusing, it looks as if the rounds have passed already. I would remove the icons and just keep empty circles or at least reduce their opacity.

File: packages/suite/src/actions/wallet/coinjoinClientActions.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/coinjoinClientActions.ts:158]: ah, right the progress, you're aright

**Review by szymonlesisz — APPROVED** [review]

**Comment by dahaca** [comment]
@komret b07ac64f032c6596e55d0ffea39896f015f97019

**Comment by komret** [comment]
nice, let's merge it!

File: packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts [hunk 3]
```diff
@@ -15,6 +17,17 @@ export const coinjoinMiddleware =
             CoinjoinBackendService.getInstances().forEach(b => b.cancel());
         }
 
+        // do not close success and critical phase modals when they are open, similar to discovery middleware
+        const { modal } = api.getState();
+        const allowedModals = ['coinjoin-success', 'critical-coinjoin-phase'];
+
+        if (
```
Comment (szymonlesisz) [inline packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts:24]:
```suggestion
       if (
            action.type === UI.CLOSE_UI_WINDOW &&
            'payload' in modal &&
            allowedModals.includes(modal.payload.type)
        ) {
            return action;
        }
```

instead of typeforcing

**Comment by szymonlesisz** [comment]
also, unit tests are failling because you need to add new dispatched action types to `__fixtures__/coinjoinClientActions`: `onCoinjoinRoundChanged` test expected results

```
result: {
            actions: [COINJOIN.SESSION_ROUND_CHANGED, "put-open-modal-constant-in-expected-position"],
            trezorConnectCalledTimes: 0,
        },
```

use `yarn workspace @trezor/suite test:unit coinjoinClientActions --coverage=false` to validate locally

File: packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts:24]: 8492c6f13b70a1c12739bbf78fd685db6024bcdd

File: packages/suite/src/actions/wallet/__fixtures__/coinjoinClientActions.ts [hunk 4]
```diff
@@ -17,11 +18,12 @@ export const onCoinjoinRoundChanged = [
         },
         params: {
             phase: 0,
-            inputs: [{ accountKey: 'b' }, { accountKey: 'c' }],
+            inputs: [{ accountKey: 'a' }, { accountKey: 'c' }],
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/__fixtures__/coinjoinClientActions.ts]: why this change? this is a valid test which checks that `COINJOIN.SESSION_ROUND_CHANGED` was not called

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/actions/wallet/__fixtures__/coinjoinClientActions.ts (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/actions/wallet/__fixtures__/coinjoinClientActions.ts]: fixed

**Review by szymonlesisz — APPROVED** [review]
